src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape:Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const isNotValid = a <= 0 || b <= 0 || c <= 0;
+    const longestSide = this.a >= b + c || this.b >= c + a || this.c >= a + b;
+
+    if (isNotValid || longestSide) {
+      // eslint-disable-next-line max-len
+      throw new Error('The incorrect value passed! The value should be bigger than 0, and the longes side cannot be bigger than the sum of the other 2 sides');
```

Comment:
```suggestion
      throw new Error('The incorrect value passed! The value should be bigger than 0,`
       + `and the longes side cannot be bigger than the sum of the other 2 sides');
```

Reply:
I leave that as I had because the suggested variant also gives the max-length error
![sugested](https://user-images.githubusercontent.com/97559412/230108709-09fe56b8-21ca-41e5-9f21-754177326dad.png)